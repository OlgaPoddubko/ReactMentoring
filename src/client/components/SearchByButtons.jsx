// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { changeSearchBy } from '../actions';

type Props = {
  searchBy: string,
  changeSearchBy: Function,
  dispatch: Function,
}

export class SearchByButtons extends React.Component<Props> {
  constructor(props) {
    super(props);
    (this:any).handleSearchByChange = this.handleSearchByChange.bind(this);
  }

  handleSearchByChange(e) {
    this.props.changeSearchBy(e.target.value);
  }

  render() {
    return (
      <React.Fragment>
        <span>search by</span>
        <button className= { this.props.searchBy == 'title' ? 'active-search search-by' : 'search-by'} type="button" value="title" onClick={this.handleSearchByChange}>title</button>
        <button className= { this.props.searchBy == 'genres' ? 'active-search search-by' : 'search-by'} type="button" value="genres" onClick={this.handleSearchByChange}>genre</button>

        <style jsx>{`
          .search-by {
            margin-left: 20px;
            padding: 10px;
            width: 100px;
            background: #4d4d4d;
            color: #ffffff;
            border: none;
            text-transform: uppercase;
          }
          .active-search {
            background: #e75f77;
          }
        `}</style>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  searchBy: state.gallery.searchBy,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  changeSearchBy,
}, dispatch);

export default connect(mapDispatchToProps)(SearchByButtons);
