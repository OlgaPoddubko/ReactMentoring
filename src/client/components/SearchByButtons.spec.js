import React from 'react';
import { SearchByButtons } from './SearchByButtons';

describe('render SearchByButtons', () => {
  it('should render SearchByButtons', () => {
    const wrapper = shallow(<SearchByButtons />);
    expect(wrapper).toMatchSnapshot();
  });
});
