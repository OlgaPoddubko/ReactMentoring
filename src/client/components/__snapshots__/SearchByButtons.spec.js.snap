// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchByButtons should render SearchByButtons 1`] = `
<React.Fragment>
  <span>
    search by
  </span>
  <button
    className="search-by"
    onClick={[Function]}
    type="button"
    value="title"
  >
    title
  </button>
  <button
    className="search-by"
    onClick={[Function]}
    type="button"
    value="genres"
  >
    genre
  </button>
  <style
    jsx={true}
  >
    
          .search-by {
            margin-left: 20px;
            padding: 10px;
            width: 100px;
            background: #4d4d4d;
            color: #ffffff;
            border: none;
            text-transform: uppercase;
          }
          .active-search {
            background: #e75f77;
          }
        
  </style>
</React.Fragment>
`;
